### PR TITLE
tests: migrating db tests to sqlalchemy

### PR DESCRIPTION
**Pull request by RustyBower**
This should fix https://github.com/sopel-irc/sopel/issues/1623

### CODE REVIEW

**Comment by dgw** [comment]
@Exirel I'd still like you to look at these changes, but I also want to merge this and #1781 together, in the next couple days, since they're _kinda_ related…ish.

File: test/test_db.py [hunk 1]
```diff
@@ -35,6 +34,7 @@
 def db():
     config = MockConfig()
     config.core.db_filename = db_filename
+    config.core.db_type = 'sqlite'
```
Comment (Exirel) [inline test/test_db.py]: I need to update #1781 so it will uses `sqlite` in its new config.

Comment (dgw) [reply test/test_db.py]: `sqlite` is already the default: https://github.com/sopel-irc/sopel/blob/185686f92d299f95027f4dccf40ffd994e7807fd/sopel/config/core_section.py#L133-L135 Did you need to update something else besides the fake config?

Comment (Exirel) [reply test/test_db.py]: Nothing else. If it's already the default, then I've nothing to update!

**Pull request merged**